### PR TITLE
specs, fix mockapi in auth scenarios

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
microsoft/typespec PR https://github.com/microsoft/typespec/pull/4740

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/authentication/api-key/mockapi.ts [hunk 1]
```diff
@@ -21,12 +21,12 @@ Scenarios.Authentication_ApiKey_valid = passOnSuccess({
...
-Scenarios.Authentication_ApiKey_invalid = passOnSuccess({
+Scenarios.Authentication_ApiKey_invalid = passOnCode(403, {
   uri: `/authentication/api-key/invalid`,
   method: `get`,
   request: {
     headers: {
-      "x-ms-api-key": "valid-key",
+      "x-ms-api-key": "invalid-key",
     },
     status: 403,
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/authentication/api-key/mockapi.ts:31]: Unsure why we had a `status: 403` in request.

**Review by msyyc — APPROVED** [review]
Context: the bug causes [dash-board](https://azure.github.io/cadl-ranch/) can't count coverage for invalid scenario for auth.
image

**Review by iscai-msft — APPROVED** [review]
approved but cn you fix the lint issues?

**Comment by weidongxu-microsoft** [comment]
Sure. Cleaned up the lint error in the repo as well.

**Pull request merged**